qulice-pmd/src/test/java/com/qulice/pmd/PMDEmptyTest.java

```diff
@@ -0,0 +1,218 @@
+            )
+        );
+    }
+
+    /**
+     * Makes sure that empty statement blocks fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptyStatementBlockTest() throws Exception {
+        this.validatePMD(
+            "EmptyStatementBlock.java",
+            Matchers.containsString(
+                "Avoid empty block statements"
+            )
+        );
+    }
+
+    /**
+     * Makes sure that empty initializers fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptyInitializerTest() throws Exception {
+        this.validatePMD(
+            "EmptyInitializer.java",
+            Matchers.containsString(
+                "Empty initializer was found"
+            )
+        );
+    }
+
+    /**
+     * Makes sure that empty statement not in a loop fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptyStatementNotInLoopTest() throws Exception {
+        this.validatePMD(
+            "EmptyStatementNotInLoop.java",
+            Matchers.containsString(
+                "An empty statement (semicolon) not part of a loop"
+            )
+        );
+    }
+
+    /**
+     * Makes sure that empty synchronized statements fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptySynchronizedBlockTest() throws Exception {
+        this.validatePMD(
+            "EmptySynchronizedBlock.java",
+            Matchers.containsString(
+                "Avoid empty synchronized blocks"
+            )
+        );
+    }
+
+    /**
+     * Makes sure that empty switch statements fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptySwitchStmtTest() throws Exception {
+        this.validatePMD(
+            "EmptySwitchStmt.java",
+            Matchers.containsString(
+                "Avoid empty switch statements"
+            )
+        );
+    }
+
+    /**
+     * Makes sure that empty finally blocks fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptyFinallyBlockTest() throws Exception {
+        this.validatePMD(
+            "EmptyFinallyBlock.java",
+            Matchers.containsString("Avoid empty finally blocks")
+        );
+    }
+
+    /**
+     * Makes sure that empty while statements fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptyWhileStmtTest() throws Exception {
+        this.validatePMD(
+            "EmptyWhileStmt.java",
+            Matchers.containsString("Avoid empty while statements")
+        );
+    }
+
+    /**
+     * Makes sure that empty if blocks fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptyIfStmtTest() throws Exception {
```

Comment:
@prahladyeri Let's call this method `failsForEmptyIfStatement`, see [this page](http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names) for more details.